pkg/azureutils/azure_disk_utils.go

```diff
@@ -111,28 +111,18 @@ func GetCachingMode(attributes map[string]string) (compute.CachingTypes, error)
 	return compute.CachingTypes(cachingMode), err
 }
 
-// GetCloudProvider get Azure Cloud Provider
-func GetCloudProvider(kubeconfig, secretName, secretNamespace, userAgent string) (*azure.Cloud, error) {
+// GetCloudProviderFromClient get Azure Cloud Provider
+func GetCloudProviderFromClient(kubeClient *clientset.Clientset, secretName string, secretNamespace string, userAgent string) (*azure.Cloud, error) {
+	var config *azure.Config
+	var fromSecret bool
+	var err error
 	az := &azure.Cloud{
 		InitSecretConfig: azure.InitSecretConfig{
 			SecretName:      secretName,
 			SecretNamespace: secretNamespace,
 			CloudConfigKey:  "cloud-config",
 		},
 	}
-
-	kubeClient, err := getKubeClient(kubeconfig)
-	if err != nil {
-		klog.Warningf("get kubeconfig(%s) failed with error: %v", kubeconfig, err)
```

Comment:
could you keep this warning `klog.Warningf("get kubeconfig(%s) failed with error: %v", kubeconfig, err)` in L190?